src/Microsoft.Security.Utilities.Core/IdentifiableSecrets.cs

```diff
@@ -341,6 +364,16 @@ public static string GenerateCommonAnnotatedTestKey(byte[] randomBytes,
             providerReserved = new byte[providerReservedLength];
         }
 
+        if (allocationTime.Kind != DateTimeKind.Utc)
+        {
+            throw new ArgumentException(nameof(allocationTime), "The allocation time must be in UTC.");
+        }
+
+        if (allocationTime.Year < 2024)
```

Comment:
>allocationTime [](http://example.com/codeflow?start=12&length=14)

We could also add the upper bounds check, i.e., if allocationTime.Year > 87

Reply:
Good idea, I will add this!

Reply:
I think the max allocation year would be 2085, right? Since it's 2024 + 61 (0 - 61 / 'A' - '9' being supported base62 digits).

Reply:
Fixed.

Reply:
right!